packages/core/src/interpreter.ts

```diff
@@ -758,6 +758,22 @@ export class Interpreter<
       );
     }
 
+    if (!events.length) {
+      const currentState = this.state;
+      const nextState: AnyState = new State({
+        ...currentState,
+        history: currentState,
+        actions: [],
+        transitions: []
+      });
+      nextState.changed = false;
+      if (nextState.history) {
+        delete nextState.history.history;
+      }
+      this._state = nextState;
+      return;
+    }
```

Comment:
We should treat sending an empty array of events as not sending any event at all:

```suggestion
    if (!events.length) {
      return;
    }
```

Reply:
This would have a somewhat weird effect though:
```ts
service.send(arr)
if (service.state.changed) {
  // oops, we can't truly trust this `.changed` value here
}
```

Reply:
> This would have a somewhat weird effect though:
> 
> ```ts
> service.send(arr)
> if (service.state.changed) {
>   // oops, we can't truly trust this `.changed` value here
> }
> ```

This makes sense though because `service.state` is still the _previous_ state, which indeed hasn't changed. The state machine should only change state in response to events (mathematically), so no transition should happen and the state should be exactly the same.

Reply:
I'm talking about a reversed scenario - where `.changed === true` even though nothing has changed:
```js
const sendAndDoSmth = (events) => {
  service.send(events)
  if (service.state.changed) {
    // do smth
  }
}

sendAndDoSmth([{ type: 'FOO' }]) // this gets handles by the machine and thus a new state has `.changed: true`
sendAndDoSmth([]) // we didn't change the internal state at all, it still has `.changed: true`

// oops, we have executed the internal conditional logic twice, although the machine's truly only has changed once
```

Reply:
Yes, that's what I mean. `service.send([])` would _not_ cause a transition, which means that it would never result in a new state anyway.

You'd be reading `.changed === true` from the _previous_ state (before the empty array was sent), which is valid and accurate.

Reply:
> You'd be reading .changed === true from the previous state (before the empty array was sent), which is valid and accurate.

I'm not sure if a common expectation after using `.send()` would be that you might get a *previous* state "back". This looks counter-intuitive to me.